endpoints/Agents.ps1

```diff
@@ -1,28 +1,33 @@
-function Get-AteraAgents {
-  return New-GetRequest -Endpoint "/agents"
-}
-
 function Get-AteraAgents {
   param(
-    # Customer ID to retrieve list of agents for
-    [int]$CustomerID
+      # Customer ID to retrieve list of agents for
+       [Parameter(Mandatory=$false)][int]$CustomerID
   )
-  return New-GetRequest -Endpoint "/agents/$CustomerID"
+  if($CustomerID){
+    return New-GetRequest -Endpoint "/agents/$CustomerID"
+  }
+  else
+  {
+    return New-GetRequest -Endpoint "/agents"
+  }
 }
 
 function Get-AteraAgent {
+  ##############
+  # If no param is given the function will get the current PC
+  ##############
   param(
     # ID of agent to retrieve
-    [int]$ID
+    [Parameter(Mandatory=$false,ParameterSetName=’ID’)][int]$ID
+    # Machine Name; Default hostname of PC
+    [Parameter(Mandatory=$false,ParameterSetName=’MachineName’)][string]$MachineName=$env:COMPUTERNAME
   )
-  return New-GetRequest -Endpoint "/agents/$AgentID"
+  if($ID){
+    return New-GetRequest -Endpoint "/agents/$AgentID"
+  }
+  if($MachineName){
+    return New-GetRequest -Endpoint "/agents/machine/$MachineName"
+  }
```

Comment:
You're using an `$AgentID` variable but not setting one... which upon further review is also what I had originally. So the `$AgentID` variable needs to be changed to `$ID`.

Reply:
Fixed in latest Push